x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/template/IndexTemplateRegistry.java

```diff
@@ -518,8 +530,14 @@ private void putComposableTemplate(
                 new ActionListener<AcknowledgedResponse>() {
                     @Override
                     public void onResponse(AcknowledgedResponse response) {
-                        creationCheck.set(false);
-                        if (response.isAcknowledged() == false) {
+                        if (response.isAcknowledged()) {
```

Comment:
That's an excellent point!! We'll need to enhance the dependencies validation mechanism to be capable to look at component templates versions as well. I think it should be doable if both the dependencies and the dependents are coming from the same registry, but not so sure otherwise...

I'll try to think of something to work around this issue

Reply:
Part 1: I extended the dependencies check that is done during index template installation to look at component templates' versions as well (see https://github.com/elastic/elasticsearch/pull/97546/commits/665dc9de880b247a590b853a3f6600579c373f60). 
Currently this check includes only component templates that are installed by the same registry that in installs the index template
This would not affect any registry that doesn't require automatic rollovers (meaning - has no effect on any other than the APM registry).

Part 2: I am now looking for a way to apply this also to component templates that come from other registries. It sorts of introduces a concept of inter-registries dependency, which I am not sure is desirable. I will try to propose something if I find a proper way. 
@jbaiera if any of you have a concrete idea, please let me know.

Reply:
@jbaiera since this plugin is going to be disabled by default for now and since the worst outcome is of premature rollover, we decided to handle "Part 2" above as a followup for this PR (see #99647). 
Please do a comprehensive review for this PR assuming that this external dependencies upgrade validation is out of scope.